kedro/ipython/__init__.py

```diff
@@ -163,7 +163,10 @@ def _resolve_project_path(
     if path:
         project_path = Path(path).expanduser().resolve()
     else:
-        if local_namespace and "context" in local_namespace:
+        if local_namespace.get("context") and hasattr(
+            local_namespace["context"], "project_path"
+        ):
```

Comment:
This is an edge case where I run `behave` and I get error, because there is a name collision where `behave` using something also call `context` in the local namespace. 

This is a generic problem though rare.